certification/internal/engine/engine.go

```diff
@@ -0,0 +1,156 @@
+		return fmt.Errorf("%w: %s", errors.ErrCreateTempDir, err)
+	}
+	log.Debug("temporary directory is", tmpdir)
+	defer func() {
+		if err := os.RemoveAll(tmpdir); err != nil {
+			log.Error("unable to clean up tmpdir", tmpdir, err)
+		}
+	}()
+
+	containerFSPath := path.Join(tmpdir, "fs")
+	if err := os.Mkdir(containerFSPath, 0755); err != nil {
+		return fmt.Errorf("%w: %s: %s", errors.ErrCreateTempDir, containerFSPath, err)
+	}
+
+	// export/flatten, and extract
+	log.Debug("exporting and flattening image")
+	r, w := io.Pipe()
+	go func() {
+		defer w.Close()
+		log.Debug("writing container filesystem to output dir", containerFSPath)
+		err = crane.Export(img, w)
+		if err != nil {
+			// TODO: Handle this error more effectively. Right now we rely on
+			// error handling in the logic to extract this export in a lower
+			// line, but we should probably exit early if the export encounters
+			// an error, which requires watching multiple error streams.
+			log.Error("unable to export and flatten container filesystem:", err)
+		}
+	}()
+
+	log.Debug("extracting container filesystem to", containerFSPath)
+	err = fileutils.Untar(containerFSPath, r)
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrExtractingTarball, err)
+	}
+
+	// store the image internals in the engine image reference to pass to validations.
+	// TODO: pass this to validations when the new check interface includes it.
+	c.imageRef = ImageReference{
+		ImageURI:    c.Image,
+		ImageFSPath: containerFSPath,
+		ImageInfo:   &img,
+	}
+
+	// execute checks
+	log.Info("executing checks")
+	for _, check := range c.Checks {
+		c.results.TestedImage = c.Image
+		targetImage := c.Image
+
+		log.Info("running check: ", check.Name())
+
+		// run the validation
+		checkStartTime := time.Now()
+		checkPassed, err := check.Validate(targetImage)
+		checkElapsedTime := time.Since(checkStartTime)
+
+		if err != nil {
+			log.WithFields(log.Fields{"result": err, "ERROR": err.Error()}).Info("check completed: ", check.Name())
+			c.results.Errors = append(c.results.Errors, runtime.Result{Check: check, ElapsedTime: checkElapsedTime})
+			continue
+		}
+
+		if !checkPassed {
+			log.WithFields(log.Fields{"result": "FAILED"}).Info("check completed: ", check.Name())
+			c.results.Failed = append(c.results.Failed, runtime.Result{Check: check, ElapsedTime: checkElapsedTime})
+			continue
+		}
+
+		log.WithFields(log.Fields{"result": "PASSED"}).Info("check completed: ", check.Name())
+		c.results.Passed = append(c.results.Passed, runtime.Result{Check: check, ElapsedTime: checkElapsedTime})
+	}
+
+	if len(c.results.Errors) > 0 || len(c.results.Failed) > 0 {
+		c.results.PassedOverall = false
+	} else {
+		c.results.PassedOverall = true
+	}
+
+	// hash contents if bundle
+	if c.IsBundle {
+		// TODO: Implement! The current implementation of this requires a podman engine
+		// and leverages shell exec calls. Leaving this out for the moment knowing we must
+		// implement before we transition.
```

Comment:
Should be a relatively simple transition. Happy to handle this once this lands.

Reply:
Perfect, thanks!